SXR/SDK/sxrsdk/src/main/jni/objects/components/transform.cpp

```diff
@@ -121,14 +121,18 @@ void Transform::setModelMatrix(glm::mat4 matrix) {
     glm::vec3 scale;
     glm::quat rotation;
     glm::vec3 translation;
-    glm::vec3 skew;
-    glm::vec4 perspective;
-    glm::decompose(matrix, scale, rotation, translation, skew, perspective);
 
     mutex_.lock();
-    position_ = translation;
-    scale_ = scale;
-    rotation_ = glm::conjugate(rotation);
```

Comment:
Keep in mind the conjugate is needed to fix issue https://github.com/Samsung/GearVRf/pull/1969. Unless you have another fix quat_cast is not going to work.